library/objective-c/EnvoyBridgeUtility.h

```diff
@@ -0,0 +1,68 @@
+#import <Foundation/Foundation.h>
+
+#import "library/common/types/c_types.h"
+
+static inline envoy_data toNativeData(NSData *data) {
+  uint8_t *native_bytes = (uint8_t *)safe_malloc(sizeof(uint8_t) * data.length);
+  memcpy(native_bytes, data.bytes, data.length);
+  envoy_data ret = {data.length, native_bytes, free, native_bytes};
+  return ret;
+}
+
+static inline envoy_data toManagedNativeString(NSString *s) {
+  size_t length = s.length;
+  uint8_t *native_string = (uint8_t *)safe_malloc(sizeof(uint8_t) * length);
+  memcpy(native_string, s.UTF8String, length);
+  envoy_data ret = {length, native_string, free, native_string};
+  return ret;
+}
+
+static inline envoy_headers toNativeHeaders(EnvoyHeaders *headers) {
+  envoy_header_size_t length = 0;
+  for (id headerKey in headers) {
+    length += [headers[headerKey] count];
+  }
+  envoy_header *header_array = (envoy_header *)safe_malloc(sizeof(envoy_header) * length);
+  envoy_header_size_t header_index = 0;
+  for (id headerKey in headers) {
+    NSArray *headerList = headers[headerKey];
+    for (id headerValue in headerList) {
```

Comment:
Can we use typed keys here instead of `id`?

Reply:
Yes. In fact, I feel like I already had changed this, but it might have gotten lost in the chain of branches.

Reply:
Ah, I did in the new location, but not the old one where I factored from.